include/Graph/Graph.hpp

```diff
@@ -85,24 +88,38 @@ class Graph {
   T_EdgeSet<T> edgeSet = {};
   std::optional<std::pair<std::string, char>> getExtenstionAndSeparator(
       InputOutputFormat format) const;
-  int writeToStandardFile(const std::string &workingDir,
-                          const std::string &OFileName, bool compress,
-                          bool writeNodeFeat, bool writeEdgeWeight,
-                          InputOutputFormat format) const;
-  int readFromStandardFile(const std::string &workingDir,
-                           const std::string &OFileName, bool compress,
-                           bool readNodeFeat, bool readEdgeWeight,
-                           InputOutputFormat format);
-  void recreateGraphFromReadFiles(
+  void writeGraphToStream(const std::ostream &oGraph,
+                          const std::ostream &oNodeFeat,
+                          const std::ostream &oEdgeWeight, const char &sep,
+                          bool writeNodeFeat, bool writeEdgeWeight) const;
+  void readGraphFromStream(const std::istream &iGraph,
+                           const std::istream &iNodeFeat,
+                           const std::istream &iEdgeWeight, bool readNodeFeat,
+                           bool readEdgeWeight);
```

Comment:
R/W with generic streams is much more portable in application (for example, use in network streams, binary streams, etc etc)